clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -2029,6 +2029,10 @@ def warn_frame_address : Warning<
   "calling '%0' with a nonzero argument is unsafe">,
   InGroup<FrameAddress>, DefaultIgnore;
 
+def warn_frame_address_missing_noinline: Warning<
+  "calling '%0' in function not marked __attribute__((noinline)) may return a caller's %1 address">,
```

Comment:
This is a minor thing, but you should probably use `%select` for `%1` here instead of passing in e.g. `"frame"` as a string. The `%0` is fine imo because it’s the name of a builtin function.